lm_eval/evaluator.py

```diff
@@ -7,11 +7,13 @@
 import lm_eval.tasks
 import lm_eval.base
 from lm_eval.utils import positional_deprecated, run_task_tests
+import transformers
 
 
 @positional_deprecated
 def simple_evaluate(
     model,
+    pre_trained_path: str,
```

Comment:
Will this break existing code which doesn't pass `pre_trained_path`? If so, make `pre_trained_path` an optional argument with a default of `None` and only conditionally pass it to `create_from_arg_string`.

Reply:
Hi, thanks for flagging I hadn't though about that. I have just checked this and I actually think it is correct as is. In the [model class](https://github.com/OpenBioML/lm-evaluation-harness/blob/master/lm_eval/models/huggingface.py), it has a statement  `assert isinstance(pretrained, str) ` so I think that a model pre-trained path must be passed in - which I guess makes sense because you would only be loading a trained model in to the evaluation pipeline. I guess this is a check to make sure you are not randomly initialising a model when you should be loading one in. We might want to change that in future if we want a random baseline, but for now maybe it is ok?

Reply:
Are you referring to [this](https://github.com/OpenBioML/lm-evaluation-harness/blob/50eb80ba5f23bde87cb80bbe1f1633bb98038e58/lm_eval/models/huggingface.py#L130) line (you can link exact lines btw)? 

I might be missing something but won't the call from `main.py` [here](https://github.com/OpenBioML/lm-evaluation-harness/blob/0244db197c03ca5883c6407482aecca3d08538a7/main.py#L79) fail since it's not passing the argument?

Reply:
I have removed pre_trained_path which returns [simple_evaluate](https://github.com/OpenBioML/lm-evaluation-harness/blob/0244db197c03ca5883c6407482aecca3d08538a7/lm_eval/evaluator.py#L6:~:text=def-,simple_evaluate,-() to its original state. It turns out that you can (and I think this is how it is supposed to be done anyway) pass in the config arg `model_args: "pretrained=pathtocheckpoint, otherargs=args"` and the [string is parsed](https://github.com/OpenBioML/lm-evaluation-harness/blob/0244db197c03ca5883c6407482aecca3d08538a7/lm_eval/evaluator.py#L6:~:text=lm%20%3D%20lm_eval.models.get_model(model).,)) into a dict of {pretrained: "pathtocheckpoint", otherargs: args}. I checked that this runs and I get the same pre-trained model loaded in. If this model_args string is missing, an error is returned which says pretrained is missing and this should be the case for both main and main_eval.